### PR TITLE
Refactor Phase 1 Metrics 

### PR DESCRIPTION
**Pull request by jiaccarino**
Description: Refactor Metrics for Phase 1
Changes:

    added metrics
    individual files for each metric
    updates to index.ts

Correct Output:
image

Work still to be done:

    Interfaces for each metric inheriting from a base class for metric
        Add 2 new metrics for phase 2

& Add Dependency Pinning https://github.com/matthewl121/CustomRegistry/issues/33

### CODE REVIEW

**Comment by glenscheng** [comment]
i see that there are 3 conflicting files

**Review by glenscheng — COMMENTED** [review]
looks good, will approve after conflicts are resolved

**Review by glenscheng — APPROVED** [review]

**Comment by matthewl121** [comment]
This pull request includes significant changes to refactor the codebase, introduce new metrics, and improve error handling. The most important changes are the migration from JavaScript to TypeScript, restructuring of metric calculations, and enhanced error logging.

### Migration to TypeScript:
* [`src/index.ts`](diffhunk://#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556L1-L110): Migrated the main logic from `src/index.js` to TypeScript, including type annotations and improved error handling. [[1]](diffhunk://#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556L1-L110) [[2]](diffhunk://#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80L1-R83)

### New Metrics Introduced:
* [`src/metrics/busFactor.ts`](diffhunk://#diff-c916f6c122f2f086aafa85d5cba6e3aad530d6dad7a808489181dc5337954711R1-R72): Added a new metric to calculate the bus factor, which assesses the risk of key contributors leaving a project.
* [`src/metrics/correctness.ts`](diffhunk://#diff-2b9fe79d3613e31d6020688859a66fa3e9b60a132344d56e285e8e0cb7170194R1-R52): Added a new metric to evaluate code correctness based on the ratio of closed to total issues.
* [`src/metrics/dependencyPinning.ts`](diffhunk://#diff-cfa93ce5707328322419ef256a7bd5338771033db0508d3d61b06390f39d0f36R1-R110): Added a new metric to check if dependencies are pinned to specific versions, enhancing security and stability.
* [`src/metrics/license.ts`](diffhunk://#diff-83fb2057e52c25291ec9771757baabf1d3f28d60859720dca0cab377ddbd24f3R1-R79): Added a new metric to verify the presence of a license in the repository, ensuring legal compliance.

File: src/index.ts [hunk 1]
```diff
@@ -1,76 +1,84 @@
...
 
+// CommonJS-style import for dotenv
+const dotenv = require('dotenv');
+
+// Load environment variables from .env file
+dotenv.config();
 
 // Function to create and manage worker threads
-export function runWorker(owner, repo, token, repoURL, repoData, metric) {
+export function runWorker(
```
Comment (matthewl121) [inline src/index.ts:14]: Good organization, format of parameters will need to change to align with eslint syntax

**Review by matthewl121 — APPROVED** [review]
Format of runWorker parameters will need to change to be in line with eslint syntax, all other files look good

**Pull request merged**